registration/include/pcl/registration/impl/sample_consensus_prerejective.hpp

```diff
@@ -120,7 +120,7 @@ void
 SampleConsensusPrerejective<PointSource, PointTarget, FeatureT>::findSimilarFeatures(
     const std::vector<int>& sample_indices,
     std::vector<std::vector<int>>& similar_features,
-    std::vector<int>& corresponding_indices)
+    std::vector<int>& corresponding_indices) const
```

Comment:
The creation of `nn_distances` can be moved down, just before `nearestKSearch` (inside `if`). Then it is only created if necessary.

Reply:
Done